src/libsyntax/parse/parser.rs

```diff
@@ -760,13 +760,19 @@ impl<'a> Parser<'a> {
     }
 
     pub fn parse_ident(&mut self) -> PResult<'a, ast::Ident> {
-        self.parse_ident_common(true)
+        self.parse_ident_common(true, false)
     }
 
-    fn parse_ident_common(&mut self, recover: bool) -> PResult<'a, ast::Ident> {
+    pub fn parse_ident_attr(&mut self) -> PResult<'a, ast::Ident> {
+        self.parse_ident_common(true, true)
```

Comment:
Could you use add logic for `Self` here in `parse_ident_attr` instead of adding one more tweak to `parse_ident_common`.
(See `parse_path_segment_ident` for an example.)